### PR TITLE
Verify OCI container images

### PR DESCRIPTION
**Pull request by deitch**
Implements OCI container image verification. Unlike ECI/VM images, which just have a sha256 hash of the file, OCI images have a single tar file with multiple layers and config inside of it.

This does the following:

1. extract the image manifest from within the tar file
1. retrieve the sha256 of the config and of each layer in the image manifest
1. calculate the sha of the config and each layer, and verify that it matches what is in the manifest
1. if a sha256 hash was provided for the OCI image in the app config from the controller, calculate the sha of the image manifest and verify that it matches what was provided in app config

It should be noted that various elements of pillar assume that `ImageSha256` _always_ will be set, and thus for containers, we set it to the image UUID. This should be changed - if there is no sha256 provided, then keep it blank - and we are working on it. In the meantime, the verifier checks if the `ImageSha256` property is a valid UUID, and therefore was overridden. If it was, it treats it as blank. This will go away when we stop overriding the `ImageSha256` property.

Also includes tests for the verification of images.

### CODE REVIEW

**Comment by deitch** [comment]
Fixed most of the yetus. I am not dealing with its complaints about tabs vs spaces in shell scripts, or its inability to handle that `cat file | sha256sum | awk ...` is valid.

File: pkg/pillar/cmd/verifier/verifier.go [hunk 1]
```diff
@@ -742,29 +741,38 @@ func markObjectAsVerifying(ctx *verifierContext,
...
 		log.Errorf("verifyObjectSha %s failed %s\n",
 			config.Name, cerr)
 		return false
 	}
-
-	got := fmt.Sprintf("%x", imageHash)
-	if got != strings.ToLower(config.ImageSha256) {
-		log.Errorf("computed   %s\n", got)
-		log.Errorf("configured %s\n", strings.ToLower(config.ImageSha256))
+	log.Infof("internal hash consistency validated for URL %s file %s", config.Name, verifierFilename)
```
Comment (eriknordmark) [inline pkg/pillar/cmd/verifier/verifier.go]: By "internal" do you mean from the root down the layers?

Comment (deitch) [reply pkg/pillar/cmd/verifier/verifier.go]: Sort of. The verifier will do two things:

1. Always take the image manifest inside the tar file, and verify that the layers and config embedded in the same file have hashes that match their file names and the hashes provided in the manifest. This is what I meant by "internal".
2. If a hash is provided by the config, check that the image manifest itself (the root) has a hash that matches that which is provided by the config.

**Comment by deitch** [comment]
Any other inputs? Or can we merge in? The yetus complains are simply wrong and should be ignored.

**Comment by rvs** [comment]
LGTM. Merging now.

**Pull request merged**